R/Plot_synergy.R

```diff
@@ -1,328 +1,332 @@
+  return(if(is.null(val)) -1 else max(val[1],2):val[2])
+}
+
+# Returns a slice of 'val' specified by 'range', as numeric.
+# If 'range' is null, 'val' is wholly returned.  The first two elements of
+# 'range' are taken to be slice endpoints; any other values are ignored.
+# The desired portion of 'val' is converted to numeric form, then returned.  An
+# error occurs if any element of the selected slice of 'val' cannot be converted
+# to a number.
+.NumericApplyRange <- function(val, range) {
+  return(as.numeric(if(is.null(range)) val else val[range[1]:range[2]]))
+}
+
+# Returns a 2D slice of 'data' specified by 'row.range' and 'col.range'.  If
+# either range is NULL, the data for the corresponding axis of 'data' is
+# returned in full.  The row.range and col.range inputs are each treated as
+# lists whose first two elements are range endpoints, and whose remaining
+# elements (if any) are ignored.
+.ApplyRanges <- function(data, row.range, col.range) {
+  # Default value is 'remove the element just after the end of the list',
+  # for each dimension of the array (a no-op that permits cleaner syntax).
+  return(data[.RangeWithDefault(row.range, default=-(nrow(data)+1)),
+              .RangeWithDefault(col.range, default=-(ncol(data)+1))])
+}
+
+# If 'range' is NULL, return 'default', otherwise return a slice.
+# The endpoints of the slice are determined by assuming 'range' is a list, and
+# picking its first two elements.  Any additional elements are ignored.
+.RangeWithDefault <- function(range, default) {
+  return(if(is.null(range)) default else range[1]:range[2])
+}
+
+
+# TODO/REVIEW: The original logic checked into GitHub at the time this function
+# was written used flipped argument order for data.coord and krig.coord.
+# Furthermore, that logic appeared to transpose its output.  Please see the
+# corresponding unit test in test_that("extended_scores_works", {...}).  The
+# original logic swapped the locations of corner values '2' and '3'.  Looking
+# at the dose/response matrices, it seems that this transposition may have been
+# a bug.  With the code below, the contour and wireframe plots are similarly
+# transposed from their current appearance in the documentation, but seem to
+# gibe better with the documented dose/response matrices.
+
+# Perform statistical interpolation to smooth plotted values.
+.ExtendedScores <- function(scores.dose, len) {
+  # len: how many values need to be predicted between two adjacent elements
+  #      of scores.dose
+  nr <- nrow(scores.dose)
+  nc <- ncol(scores.dose)
+  # Note, not all kriging implementations are suitable for all purposes.  In
+  # particular, SpatialExtremes offers an implementation that works with small
+  # data sets.  Earlier versions of this implementation that used other kriging
+  # packages only worked well with large ones.
+  return(SpatialExtremes::kriging(
+    data = c(scores.dose),
+    data.coord = cbind(rep(1:nc, nr), rep(1:nr, each = nc)),
```

Comment:
change to data.coord =  cbind(rep(1:nr,  nc), rep(1:nc, each = nr))

Reply:
Done. The unit test passed as-is with this change, meaning that the unit test does not evaluate whether data.coord is correct. If there's a way to enhance the test to evaluate that (so that it fails if data.coord is wrong), it would be great to include it.